.github/workflows/docs.yml

```diff
@@ -17,13 +18,16 @@ permissions:
 
 jobs:
   build:
+    if: github.event.pull_request.merged == true
     environment:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
 
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+      with:
+        ref: "main"
```

Comment:
probably not necessary since we only ever merge PRs to main

Reply:
True, but here I think it has value to document what's happening. Also, it ensures that the github.io docs are only generated from main.

Reply:
is it not also redundant with the `branches: main` above?

Reply:
yes, it's unneeded. but it's not going to slow anything down. It would be better if there was a single variable for the same place, I'm checking now if yaml can do that, but I feel safer (probably irrationally) that having both in place could prevent issues in the future.

Reply:
Anchors don't work. I can remove it, but I would rather leave it in.

Reply:
By anchors, i mean the yaml feature that would automatically repeat a value, so main is in this twice.